[Hibeom0929] iP

Hibeom0929 opened the pull request:
# UnivusPro
> "Your mind is for having ideas, not holding them." - David Allen ([source](https://dansilvestre.com/productivity-quotes/))

UnivusPro frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,
1. download if from [here](https://github.com/Hibeom0929/ip/releases/tag/A-Jar).
2. double-click it.
3. add your tasks
4. let it manage your tasks for you :D

And it is **FREE**!

Features:
- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon)

#
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
    public static void (String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Hibeom0929 committed:
Level-0

Hibeom0929 committed:
Level-1 Echo

Hibeom0929 committed:
Level 2. Add, List

Hibeom0929 committed:
Level 3. Mark as Done

Hibeom0929 committed:
Level-4 ToDos, Events, Deadlines

Hibeom0929 committed:
A-TextUiTesting

Hibeom0929 committed:
Level 5. Handle Errors

Hibeom0929 committed:
Level 6. Delete

Hibeom0929 committed:
Level 7. Save

Hibeom0929 committed:
Merge branch 'branch-Level-7'

yiwern5 commented on src/main/java/Deadlines.java:
```diff
@@ -0,0 +1,13 @@
+public class Deadlines extends Task {
```
Is it better to use a singular instead of plural noun for the class name? (same goes for ToDos and Events, considering Task is in singular)

yiwern5 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,32 @@
+public class Task {
+    protected String description;
+    protected boolean isDone;
```
Good job in correctly naming the boolean

yiwern5 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,32 @@
+        return isDone;
+    }
+
+    public void unMark() {
```
Nice attempt at using camel case, but I believe 'unmark' should be treated as a single term

yiwern5 commented on src/main/java/ToDos.java:
```diff
@@ -0,0 +1,9 @@
+public class ToDos extends Task{
```
Task should be followed by a whitespace
```suggestion
public class ToDos extends Task {
```

yiwern5 commented on src/main/java/Univus.java:
```diff
@@ -0,0 +1,229 @@
+import java.io.*;
```
It would be preferable to avoid using wildcard imports

yiwern5 commented on src/main/java/UnivusException.java:
```diff
@@ -0,0 +1,5 @@
+public class UnivusException extends Throwable{
```
Throwable should be followed by a whitespace
```suggestion
public class UnivusException extends Throwable {
```

yiwern5 commented on src/main/java/Events.java:
```diff
@@ -0,0 +1,14 @@
+public class Events extends Task{
```
Task should be followed by a whitespace
```suggestion
public class Events extends Task {
```

yiwern5 commented on src/main/java/Univus.java:
```diff
@@ -0,0 +1,229 @@
+                    throw new UnivusException("OOPS!!! Wrong Index!");
+                }
+                Task task = store.get(index - 1);
+                if (task.isDone()) {
```
I appreciate how you handled the situation where users should be unable to mark the task as done when it is already marked as done, same goes for unmark

yiwern5 replied:
I appreciate how you handled the situation where users should be unable to mark the task as done when it is already marked as done, same goes for unmark

yiwern5 reviewed (commented):
LGTM! Overall, your code is clean and easy to understand. The main class is quite long, maybe can consider making it more OOP on your subsequent increment! I've added some suggestions for you too 😊

Hibeom0929 committed:
Level 8. Dates and Times

Hibeom0929 committed:
Merge branch 'branch-Level-8'

Hibeom0929 committed:
Make the code more OOP

Hibeom0929 committed:
A-More-OOP

Hibeom0929 committed:
Merge branch 'A-MoreOOP'

Hibeom0929 committed:
A-Packages

Hibeom0929 committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2324S2/ip

Hibeom0929 committed:
A-JUnit

Hibeom0929 committed:
Merge branch 'A-JUnit'

tanjiajiajun reviewed (commented):
Hi, I am tasked to do a PR review on your code. I will be reviewing based on code quality guidelines

1. In your `Parser.java`, I see that you have repeated 
```
System.out.println("____________________________________________________________");
``` 
a lot, maybe you can consider creating a variable `private final String LINE`
2. In `Storage.java`, you have 
```
int endIndex = message.lastIndexOf("(");
int startIdx = message.indexOf(":");
```
It would be nice if you can standardise if you want to name them Index or Idx.

Other than that, the rest of the code looks good! LGTM.

Hibeom0929 committed:
A-JavaDoc

Hibeom0929 committed:
A-CodingStandard

Hibeom0929 committed:
Level-9 Find

Hibeom0929 committed:
Merge branch 'A-JavaDoc'

Hibeom0929 committed:
Merge branch 'A-CodingStandard'

# Conflicts:
#	src/main/java/univus/Parser.java
#	src/main/java/univus/Ui.java

Hibeom0929 committed:
Merge branch 'Level-9'

# Conflicts:
#	src/main/java/univus/task/TaskList.java

Hibeom0929 committed:
A-CheckStyle

Hibeom0929 committed:
Merge branch 'A-CheckStyle'

Hibeom0929 committed:
GUI

Hibeom0929 committed:
Merge branch 'Level-10'

Hibeom0929 committed:
Fixed SLAP issue in Parser class and Storage class. Also added correct javadoc for several methods.

Hibeom0929 committed:
The code for marking and unmarking tasks lacks explicit checks to ensure that the tasks are correctly marked or unmarked.

The mark and unmark methods have no ability to catch potential issues during development.

Assert statements are useful to validate the expected state of tasks after marking or unmarking.

Let's add assert statements to verify the correctness of task marking and unmarking.

The assert statements serve as a form of self-documentation and provide a safety net for identifying and addressing potential issues related to task state.

Hibeom0929 committed:
Different logics are attached together in mark and unmark methods.

It causes less readability and makes it not easy to understand codes logically.

Let's add white spaces between different logics.

Having white spaces between different logics can make it clear to read logically.

Hibeom0929 committed:
Merge pull request #2 from Hibeom0929/heads/A-Assertions

Add assertions to mark and unmark methods

Hibeom0929 committed:
Merge pull request #3 from Hibeom0929/heads/A-CodeQuality

Improvement for the code quality and remove unused import and local variables.

Hibeom0929 committed:
Detecting whether there are the same task already in the current TaskList, then throw an exception.

Hibeom0929 committed:
Merge branch 'C-DetectDuplicates'

Hibeom0929 committed:
Set title to "Univus"

Hibeom0929 committed:
Fixed bugs in tests

Hibeom0929 committed:
Create file if there is no file

Hibeom0929 committed:
Update README.md

Hibeom0929 committed:
Update README.md

Hibeom0929 committed:
adjust the hard write path issue

Hibeom0929 committed:
resize the dialog box
